File: server/api/v1/shop/shop_category.go
package shop

import (
	"github.com/flipped-aurora/gin-vue-admin/server/global"
	"github.com/flipped-aurora/gin-vue-admin/server/model/common/response"
	"github.com/flipped-aurora/gin-vue-admin/server/model/shop"
	shopReq "github.com/flipped-aurora/gin-vue-admin/server/model/shop/request"
	"github.com/flipped-aurora/gin-vue-admin/server/service"
	"github.com/gin-gonic/gin"
	"go.uber.org/zap"
)

type ShopCategoryApi struct{}

var shopCategoryService = service.ServiceGroupApp.ShopServiceGroup.ShopCategoryService

func (a *ShopCategoryApi) CreateShopCategory(c *gin.Context) {
	var category shop.ShopCategory
	if err := c.ShouldBindJSON(&category); err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	if err := shopCategoryService.CreateShopCategory(&category); err != nil {
		global.GVA_LOG.Error("创建失败!", zap.Error(err))
		response.FailWithMessage("创建失败", c)
	} else {
		response.OkWithMessage("创建成功", c)
	}
}

func (a *ShopCategoryApi) DeleteShopCategory(c *gin.Context) {
	ID := c.Query("ID")
	if ID == "" {
		response.FailWithMessage("参数错误", c)
		return
	}
	if err := shopCategoryService.DeleteShopCategory(ID); err != nil {
		global.GVA_LOG.Error("删除失败!", zap.Error(err))
		response.FailWithMessage(err.Error(), c)
	} else {
		response.OkWithMessage("删除成功", c)
	}
}

func (a *ShopCategoryApi) UpdateShopCategory(c *gin.Context) {
	var category shop.ShopCategory
	if err := c.ShouldBindJSON(&category); err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	if err := shopCategoryService.UpdateShopCategory(category); err != nil {
		global.GVA_LOG.Error("更新失败!", zap.Error(err))
		response.FailWithMessage("更新失败", c)
	} else {
		response.OkWithMessage("更新成功", c)
	}
}

func (a *ShopCategoryApi) FindShopCategory(c *gin.Context) {
	ID := c.Query("ID")
	if category, err := shopCategoryService.GetShopCategory(ID); err != nil {
		global.GVA_LOG.Error("查询失败!", zap.Error(err))
		response.FailWithMessage("查询失败", c)
	} else {
		response.OkWithData(gin.H{"data": category}, c)
	}
}

func (a *ShopCategoryApi) GetShopCategoryList(c *gin.Context) {
	var pageInfo shopReq.ShopCategorySearch
	if err := c.ShouldBindQuery(&pageInfo); err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	if list, total, err := shopCategoryService.GetShopCategoryList(pageInfo); err != nil {
		global.GVA_LOG.Error("获取失败!", zap.Error(err))
		response.FailWithMessage("获取失败", c)
	} else {
		response.OkWithDetailed(response.PageResult{
			List:     list,
			Total:    total,
			Page:     pageInfo.Page,
			PageSize: pageInfo.PageSize,
		}, "获取成功", c)
	}
}

func (a *ShopCategoryApi) GetCategoryTree(c *gin.Context) {
	if tree, err := shopCategoryService.GetCategoryTree(); err != nil {
		global.GVA_LOG.Error("获取失败!", zap.Error(err))
		response.FailWithMessage("获取失败", c)
	} else {
		response.OkWithData(gin.H{"list": tree}, c)
	}
}

File: web/src/api/shopCategory.js
import service from '@/utils/request'

// @Tags ShopCategory
// @Summary 创建商品分类
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.ShopCategory true "创建商品分类"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"创建成功"}"
// @Router /shopCategory/createShopCategory [post]
export const createShopCategory = (data) => {
  return service({
    url: '/shopCategory/createShopCategory',
    method: 'post',
    data
  })
}

// @Tags ShopCategory
// @Summary 删除商品分类
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.ShopCategory true "删除商品分类"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /shopCategory/deleteShopCategory [delete]
export const deleteShopCategory = (params) => {
  return service({
    url: '/shopCategory/deleteShopCategory',
    method: 'delete',
    params
  })
}

// @Tags ShopCategory
// @Summary 更新商品分类
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.ShopCategory true "更新商品分类"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /shopCategory/updateShopCategory [put]
export const updateShopCategory = (data) => {
  return service({
    url: '/shopCategory/updateShopCategory',
    method: 'put',
    data
  })
}

// @Tags ShopCategory
// @Summary 用id查询商品分类
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query model.ShopCategory true "用id查询商品分类"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /shopCategory/findShopCategory [get]
export const findShopCategory = (params) => {
  return service({
    url: '/shopCategory/findShopCategory',
    method: 'get',
    params
  })
}

// @Tags ShopCategory
// @Summary 分页获取商品分类列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query request.PageInfo true "分页获取商品分类列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /shopCategory/getShopCategoryList [get]
export const getShopCategoryList = (params) => {
  return service({
    url: '/shopCategory/getShopCategoryList',
    method: 'get',
    params
  })
}

// @Tags ShopCategory
// @Summary 获取分类树
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query request.PageInfo true "获取分类树"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /shopCategory/getCategoryTree [get]
export const getCategoryTree = (params) => {
  return service({
    url: '/shopCategory/getCategoryTree',
    method: 'get',
    params
  })
}
